provider/rosa_operator_roles_data_source.go

```diff
@@ -0,0 +1,230 @@
+	response *tfsdk.ReadDataSourceResponse) {
+	// Get the state:
+	state := &RosaOperatorRolesState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	stsOperatorRolesList, err := t.awsInquiries.STSCredentialRequests().List().Send()
+	if err != nil {
+		t.logger.Error(ctx, "Failed to get operator list")
+		return
+	}
+
+	stsOperatorMap := make(map[string]*cmv1.STSOperator)
+	stsOperatorRolesList.Items().Each(func(stsCredentialRequest *cmv1.STSCredentialRequest) bool {
+		// TODO: check the MinVersion of the operator role
+		t.logger.Debug(ctx, "Operator name: %s, namespace %s, service account %s",
+			stsCredentialRequest.Operator().Name(),
+			stsCredentialRequest.Operator().Namespace(),
+			stsCredentialRequest.Operator().ServiceAccounts(),
+		)
+		// The key can't be stsCredentialRequest.Operator().Name() because of constants between
+		// the name of `ingress_operator_cloud_credentials` and `cloud_network_config_controller_cloud_credentials`
+		// both of them includes the same Name `cloud-credentials` and it cannot be fixed
+		stsOperatorMap[stsCredentialRequest.Operator().Namespace()] = stsCredentialRequest.Operator()
+		return true
+	})
+
+	get, err := t.clustersClient.Cluster(state.ClusterID.Value).Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't find cluster",
+			fmt.Sprintf(
+				"Can't find cluster with identifier '%s': %v",
+				state.ClusterID.Value, err,
+			),
+		)
+		return
+	}
+	object := get.Body()
+	sts, ok := object.AWS().GetSTS()
+	if ok {
+		accountRolePrefix := DefaultAccountRolePrefix
+		if !state.AccountRolePrefix.Unknown && !state.AccountRolePrefix.Null {
+			accountRolePrefix = state.AccountRolePrefix.Value
+		}
+
+		// TODO: use the sts.OperatorRolePrefix() if not empty
+		// There is a bug in the return value of sts.OperatorRolePrefix() - it's always empty string
+		for _, operatorRole := range sts.OperatorIAMRoles() {
+			r := OperatorIAMRole{
+				Name: types.String{
+					Value: operatorRole.Name(),
+				},
+				Namespace: types.String{
+					Value: operatorRole.Namespace(),
+				},
+				RoleARN: types.String{
+					Value: operatorRole.RoleARN(),
+				},
+				RoleName: types.String{
+					Value: getRoleName(state.OperatorRolePrefix.Value, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				PolicyName: types.String{
+					Value: getRoleName(accountRolePrefix, operatorRole.Namespace(), operatorRole.Name()),
+				},
+				ServiceAccounts: getServiceAccount(stsOperatorMap[operatorRole.Namespace()].ServiceAccounts(), operatorRole.Namespace()),
+			}
+			state.OperatorIAMRoles = append(state.OperatorIAMRoles, &r)
+		}
+	}
+	// Save the state:
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+// TODO: should be in a separate repo
+func getRoleName(prefix string, namespace string, name string) string {
```

Comment:
I think `buildSpecialName` is better here, because it called for both Role and Policy

Reply:
Changed it to be as it called in rosa